### DISCUSSION TITLE
Design concepts for PR #44770

### DISCUSSION THREAD

**Discussion by jameskoster** [WordPress/gutenberg#45346] [General]
The design discussion in #44770 was getting a bit distracting, let's bring that conversation here so that the PR can focus on implementation. 

Getting the PR into a good spot is tricky because we need to concentrate not only on its immediate focus, but also keep one eye on the future upgrades we have in mind, most notably:

1. The addition of intrinsic browsing in the 'frame'.
2. Moving lists (like Templates and Template Parts in the short term, and Content in the future) to the management area as drilldowns. This will be critical in getting the mobile experience right and is a concession we may have to make in 44770.
3. Adding Style and Navigation management to the management area.

All that said, here's the latest design concept we have:

https://user-images.githubusercontent.com/846565/198273256-da088d5f-5ffb-436e-bf6a-0728c24da3ea.mp4

The trickiest part to get right seems to be the Edit button, particularly how it connects to the frame. 

Besides this we also need to finalise design details like; spacing around the frame, colors/dimensions of the management area, including some visual separation of content items.

**Comment by jasmussen** [comment]
Thank you for sharing, this is coming together well. I think the larger frame can work to add space for controls, including those you added at the bottom. It would also be interesting to have the edge of the frame surface a resize cursor, so you can manually scale down, or up and snap into "edit mode".

The Edit button, the placement of it, and the controls outside, remains one of the bigger challenges. I think the current placement works, as it has context to the frame itself. Though I'd omit as many controls as possible — it's always easy to add back once it's clear how necessary they are, but it's hard if they are there from the start. For example, I'd omit the home icon, and the external link button. 

The "pizzaplanet.com" URL shown in context of the Edit button adds context that makes me think this _could_ potentially sit geographically elsewhere, and still work, if for example we wanted the frame to be closer to the edge. This would likely be useful for cases where the frame will be used to show list views, such as templates or otherwise.

Should we put Pages and Blog under a "Content" drilldown?

---

I also took #44770 for a spin and have some feedback on that PR specifically. Here's what it looked like in my testing:

![status](https://user-images.githubusercontent.com/1204802/198277838-c58605a9-db74-44c1-8e93-7267aec66cb3.gif)

Mainly the simpler navigation by removing the Browse/Editor links present in earlier versions, already improves things.

There's an awkward focus style around the site logo/w button:
<img width="236" alt="Screenshot 2022-10-27 at 11 30 37" src="https://user-images.githubusercontent.com/1204802/198278000-5088c7d1-7167-4544-aa36-4c1c52350ee7.png">

Probably the most important next step to figure out, is where to put the edit button if not inside a faux-browser-bar like this one:
<img width="1007" alt="Screenshot 2022-10-27 at 11 30 43" src="https://user-images.githubusercontent.com/1204802/198278056-ec41b1a1-343e-44bc-a41b-eaeb2d623ec7.png">

As noted, I think above can work. Probably also to the left can work. 

<img width="170" alt="Screenshot 2022-10-27 at 11 30 47" src="https://user-images.githubusercontent.com/1204802/198278230-c8fa7b62-c9da-4dd7-a833-ac53b3be5d6c.png">

In your prototype, this back chevron sits only on drilldown pages, not on the main page. I.e. it's not a back to dashboard arrow. Should it be?

Speaking of, there are some immediate metric things we can start working on in the PR, to match your prototype here:

<img width="613" alt="Screenshot 2022-10-27 at 13 59 10" src="https://user-images.githubusercontent.com/1204802/198278575-d73ab212-542a-4ac1-8b42-f0e4fb123ce8.png">

* The width of the sidebar
* The alignment of the chevron under the W logo, and the wider indent of other menu items
* The button metrics and colors

PR for comparison:

<img width="281" alt="Screenshot 2022-10-27 at 11 30 52" src="https://user-images.githubusercontent.com/1204802/198278720-7303e886-5add-4db5-86cc-4e7ecd1d4fd7.png">

**Reply by youknowriad** [reply]
I tried improving the alignments and colors of the sidebar in the last commit. What sidebar width should I be using?

**Reply by robglidden** [reply]
> Should we put Pages and Blog under a "Content" drilldown?

For UX, I'd suggest exploring organizing by a site's actual, or user-perspective, hierarchy.

IE the good old WordPress template hierarchy, which, on reflection, is the site hierarchy.  From top-down, 6 site areas:

Homepage
Pages
Posts
Archives
Not Found
Search

Probably a very user-intuitive list. 

And right there, just not so obviously, in the traditional template hierarchy [chart](https://developer.wordpress.org/themes/basics/template-hierarchy/), though that was drawn in a more bottom-up classic theme-programmer way, rather than a user's site editing viewpoint.

Benefit: the site editor UX as it builds out and drills down, will work with, rather than hide/reorganize/rename/fight, the site architecture.  And leverage the user-intuitive aspects of the still-essential template hierarchy.

**Reply by jasmussen** [reply]
> I tried improving the alignments and colors of the sidebar in the last commit. What sidebar width should I be using? 

Try 360px?

**Reply by youknowriad** [reply]
I've used 320 as 360 felt a bit too big for me. you can tweak the number to try.

**Comment by jameskoster** [comment]
On the subject of resizing, we should decide how template parts appear in the frame. If they match the editor we get the following:

<img width="1199" alt="Screenshot 2022-10-27 at 14 20 59" src="https://user-images.githubusercontent.com/846565/198295802-3343dcb1-4ba4-4b32-92c2-8511f4f5adf9.png">

The resize handles are awkward because the canvas is non-interactive. But in general, the 'focussed' appearance (dark grey space around the template part) feels unnecessary, as does the full-height frame. 

Here's an alternative, where the frame wraps the template part, and the resize handles appear outside:

![template-part](https://user-images.githubusercontent.com/846565/198297924-450d4ebb-e8e4-45cd-b011-62396a202c6c.gif)

And here's the concept applied when a template (or item of content) is visible in the frame:

![template](https://user-images.githubusercontent.com/846565/198299013-9b60cc21-1c74-4e43-958d-a3526aff0842.gif)

**Reply by jasmussen** [reply]
Oh I like when the frame itself becomes the canvas. Makes me wonder if those resize handles could exist also next to the default frame, but perhaps be invisible until the cursor is near.

**Reply by jameskoster** [reply]
Do you mean like in the third gif? Or are you suggesting that the resize handles persist in both browse and edit views?

**Reply by jasmussen** [reply]
Yes, in the 3rd gif exactly, but with the resize handles invisible until your cursor is near the browser edge.

**Reply by mtias** [reply]
We also should consider if the handle just changes the split between sidebar / canvas or if it only resizes the frame. I think we should try having only one handle and a bit more separated from the frame.

**Comment by jameskoster** [comment]
> The Edit button, the placement of it, and the controls outside, remains one of the bigger challenges

Let's explore some options:

<img width="630" alt="Screenshot 2022-10-27 at 14 55 06" src="https://user-images.githubusercontent.com/846565/198304248-a5ea64fa-d1a8-480b-a1cb-5cf47e3bf2e8.png">

Figma [here](https://www.figma.com/file/wXU0x4xZUuiNmLrtCWfFZW/Browse-view?node-id=1093%3A32145).

**Reply by jasmussen** [reply]
I like aspects of all of these, nice. Just to try out an additional option, here are on-hover controls:

![hover](https://user-images.githubusercontent.com/1204802/198536354-cd07b264-5aaa-490b-aa66-2e203b46afbd.gif)

**Reply by jameskoster** [reply]
I kind of ruled out having anything appear 'floating' above the frame for two reasons:

1. Discoverability – the edit action is critical, I'm not very comfortable 'hiding' it like this
2. At some point it will inevitably obscure something you want to actually click on (when we add browse-ability), which would be really frustrating.

It's a shame because it looks nice, and I appreciate the noise reduction.

**Reply by youknowriad** [reply]
I've moved the edit button and title outside the canvas in the PR. Let me know how it feels.

**Reply by jasmussen** [reply]
Took it for a spin:

![status](https://user-images.githubusercontent.com/1204802/199006833-9464f273-4d29-4c9f-bb2e-c582a1b401b9.gif)

This actually works pretty well. There's some design work to find to make it feel totally right — something between what Jay has and this — but in practice, it feels nice. I like the context of the edit button to the frame, even if, as noted, the design isn't totally right yet.

**Reply by jameskoster** [reply]
Thinking a little further down the road and considering other elements we'll need to tackle (like the admin bar), I have another concept to share for the Edit button:

https://user-images.githubusercontent.com/846565/200344582-d51a82eb-00cf-4999-b22e-48eed7f046c2.mp4

This demonstrates how folks can move between viewing the frontend of their site to editing / managing, and back again. I appreciate we'd need loading states between the frontend and editor/browse view, but it gives a general flavor.

In terms of the PR, the changes would be:

* Update site title to link to the frontend
* Include the entity name in the edit button label (this is important because the drilldown panel and browse frame may not always be in sync)
* Position edit button adjacent to site title
* Display menu icon when hovering the W button in the Edit view
* Style / spacing polish

What do y'all think?

**Reply by jasmussen** [reply]
I think there's a lot to like bout the frontend inclusion as you've explored, but that this particular aspect is probably something to hone for the future.

Just for the PR in question, though, I like the "Edit Index template" button idea. Can it just be "Edit Index"? We'll probably need all the space we can get if it is to sit in the sidebar on the left. 

I don't love how the editor bar animates in from the top, I feel like we had some better patterns earlier on. Mainly it feels like it's a curtain that sits separatly from the W logo. I'm blissfully unaware of how hard this would be to animate in practice, so feel free to shoot this down, but what if the viewport was a _mask_ that when it scaled up simply revealed that the editor bar _was there all along_?

**Reply by jameskoster** [reply]
> I think there's a lot to like bout the frontend inclusion as you've explored, but that this particular aspect is probably something to hone for the future.

I agree, but still think it's a useful dot to connect, since we're going to have to tackle the admin bar sooner or later. If we keep one eye on that now we can avoid making trouble for ourselves in the future.

> Can it just be "Edit Index"

_Potentially_, yes. My reasoning for the 'template' suffix is that the context is useful when templates have ambiguous names (e.g. 'Edit Category'), or names that are easily confused with content (e.g. 'Edit Home').

> I don't love how the editor bar animates in from the top

Aside from the animation which we can tweak, the detachment was my interpretation of Pablo's comment [here](https://github.com/WordPress/gutenberg/pull/44770#issuecomment-1289580712), particularly this part:

> The editor top bar (editor header) shall move independently when in transition to browse mode, as a distinct material that doesn’t belong to the canvas, it is a control device that rather pertains to the system that helps me manipulate my site.

This is an important element to get right because it seems it is not trivial to detach. Did you have any other thoughts on this?

**Reply by jasmussen** [reply]
I'll keep thinking about the animation 👍 👍

**Comment by robglidden** [comment]
> Should we put Pages and Blog under a "Content" drilldown?

A little more broadly:  which (and how many) top and second-level items should be in the site management sidebar?

There are many possibilities, below are a few.

## Working assumption (6)

current items (Templates, Template Parts) + "stuff"

Styles
Navigation
Templates
Template Parts
Library
Content

Assumptions/Observations

- The WordPress Template Hierarchy could be an intuitive starting point for organizing templates.
- Posts and Pages are natural orderings for navigating  both Templates and Content, and could be combined, with UX effort.
- The current homepage is the default thing shown in the Site area.  
- The current homepage, somewhat unintuitively, could be a template or a page (+ a template), depending on Settings > Reading.
- 2nd level items could be more complex that just a link, and the top bar could show additional info.
- 7-8 items is on the high side. The current 10 default admin sidebar items (before plugins) have 0 to 8 2nd level items. 

Dashboard 2
Posts 4
Media 2
Pages 2
Comments 0
Appearance 2
Plugins 2
Users 3
Tools 8
Settings 7

So here are some opinionated "bests" for 5, 6, 7 items at top level:

## Best 5 Top Level Items

Styles
Navigation
Templates
Library
Content

Benefit:  Tells a clear short story to a user
Downside: Template Parts under Templates or Library

## Best 6 Top Level Items

Homepage
Styles
Navigation
Templates
Library
Content

or

Styles
Navigation
Templates
Library
Pages
Posts

Benefits:  

- Either better than Template Parts at top level
- Homepage item could clarify default view

## Best 7 Top Level Items

Homepage
Styles
Navigation
Pages
Posts
Archives
Library

Benefit:  

- A clear story that maximizes top level
- Aligns to site architecture of existing template hierarchy

The most different from the current, but maybe my favorite.

## Possible second-level items

Content (2)
Library (3) with template parts
Templates (6), 4 if Search, Not Found moved to pages

Content

- Posts
- Pages

Templates

- Homepage
- Pages
- Posts
- Search

Library

- Reusable blocks
- Patterns
- Template parts

## Expanded Template Hierarchy (6 areas)

Homepage

- Default
- Home
- Front Page
- Page
- Latest posts

Pages

- Default
- Custom Post Types
- User-Created

Posts

- Default
- User-Created

Archives

- Default
- Author
- Category
- Custom Post Type
- Taxonomy
- Date
- Tag

Not Found
Search

**Comment by jasmussen** [comment]
Taking the latest state of the PR for a spin, and I'd love for us to move towards a state where we can make a decision on what to merge as a first iteration. Here's a snapshot of the current status:

![status](https://user-images.githubusercontent.com/1204802/199453425-0541a1f5-9160-4110-845f-142971e6b323.gif)

Working well: 

* The general layout of the sidebar, which is close to the design Jay has in Figma in dimensions.
* The animation moving in and out of edit mode. 
* The `scale` being applied to contents inside the frame — items still feel clickable in the overview (I know they aren't yet, but that was my main concern).

A few things that remain unclear. One being: I don't think we should show the back button here, it should be present only for menu drilldowns, lest when you click-click-click from a deep nesting you don't accidentally go to the dashboard. I think it's best to have the W menu be the "back to dashboard" affordance:

<img width="234" alt="Screenshot 2022-11-02 at 10 34 39" src="https://user-images.githubusercontent.com/1204802/199455206-487bc03a-7010-40de-b10e-ed7bcfc0e59f.png">

Details territory, it would be nice if the icons could line up better with the D in "Design". Jay's mockups have them a bit closer:

<img width="45" alt="Screenshot 2022-11-02 at 10 34 50" src="https://user-images.githubusercontent.com/1204802/199455317-e91a22aa-803c-4b6e-a19b-f47ba8f882ff.png">

There's still an open question on how to treat template parts in the browse mode. A starting point might be to just hide the resize handles in this overview mode:

<img width="1392" alt="Screenshot 2022-11-02 at 10 33 10" src="https://user-images.githubusercontent.com/1204802/199455384-1c232e45-9391-4551-a704-f6c376e5cf92.png">

While the animation looks good, I'm not sure about the apperance of the editor bar, mainly because it causes a jump in the content. It's not jarring, to be fair, but this is something to decide on:

![toolbar](https://user-images.githubusercontent.com/1204802/199455650-ef7ecf39-7d94-4330-92b9-1df65cf2e919.gif)

Should the animation be a bit faster? I always like for animation to be very fast, never longer than 0.2s.

One big challenge of course is the Edit button, where it should sit. I personally think either of these three options, based on Jay's work, can work:

<img width="1787" alt="options" src="https://user-images.githubusercontent.com/1204802/199457192-ebe68cbb-e7cf-437f-b517-9c1a1b9d0db2.png">

First and second option are close to what's in the branch now, but require a larger clearance around the frame.

Third option is pretty simple, but will likely work as well, even if it loses some context. This might work better with a wider sidebar, in the PR it's 320px vs. 360px in Jay's mockups. 

But ultimately, whether we go one route or the other, I have a feeling this boils down to some of the technical aspects. It's important to consider how the contents of the frame shift from the Templates list, and to the browse view. Right now it's a bit jumpy, and in fact the dimensions of the frame change a bit:

![transition](https://user-images.githubusercontent.com/1204802/199457613-a9cec446-4102-478a-b02b-6e65636444ca.gif)

That isn't ideal, and this shifting is something we need to feel pretty good about on the technical level, if the frame is going to become such a large cornerstone of the design. What are our options here for making this feel resilient?

* Does it simplify things if the Edit button sits on the left vs. above?
* Does it simplify things if the frame always has identical clearing, even if large, all around?
* Can the frame be animated when it changes size? Does the way content is loaded inside afford this?

I know these are rather big questions, but they are important to get right before we merge this, and would love also @pablohoneyhoney or @mtias insights.

**Reply by jameskoster** [reply]
> The animation moving in and out of edit mode.

There's an issue with the implementation currently. It scales the contents of the iframe rather than the iframe itself, which harms the impression of this being a zoomed out view. It's particularly damaging when full-width blocks are visible:

| Full-width featured image in editor | Full-width featured image in browse frame |
| --- | --- | 
| <img width="823" alt="Screenshot 2022-11-01 at 11 25 18" src="https://user-images.githubusercontent.com/846565/199473201-1457a3f4-472f-4589-99cf-fe94376f8351.png"> | <img width="821" alt="Screenshot 2022-11-01 at 11 25 26" src="https://user-images.githubusercontent.com/846565/199473204-2ef82470-75f9-421e-8fa2-7e2d42cc2ae9.png"> | 

It would be better to aim for something like this, which scales only the iframe:


https://user-images.githubusercontent.com/846565/199474845-ed627290-3bc8-46c3-af2a-1e55490898e4.mp4



> A starting point might be to just hide the resize handles in this overview mode

I think removing the padding and height would have a more significant impact:

| Remove padding and height | Current state of PR |
| --- | --- |
| <img width="1288" alt="Screenshot 2022-11-02 at 11 05 36" src="https://user-images.githubusercontent.com/846565/199474498-91e3844e-59c1-4c9c-8d96-07621f2ec735.png"> | <img width="1287" alt="Screenshot 2022-11-02 at 11 06 42" src="https://user-images.githubusercontent.com/846565/199474500-d9101e24-2406-44a1-8a2b-2be19e94f69a.png"> |



> Third option is pretty simple, but will likely work as well, even if it loses some context

I feel fairly strongly that the context is important. If not right now, then when we add browse-ability to the frame and it becomes possible for the left-hand navigation area and contents of the frame to become navigationally detached from one another. That said, printing the url/template name here is only one option. Another could be for the edit mode to [better clarify exactly what it is you're editing](https://github.com/WordPress/gutenberg/issues/45461). All that to say, maybe just the Edit button on its own is fine to begin with.

**Reply by youknowriad** [reply]
Some updates:

 - I removed the back button
 - I updated the "template parts" view mode per @jameskoster suggestions

> Does it simplify things if the Edit button sits on the left vs. above?

I don't think this changes a lot of things. 

> Does it simplify things if the frame always has identical clearing, even if large, all around?

The space can differ from side to another but I find that I prefer if it stays the same across pages (list view, editor...)

> There's an issue with the implementation currently. It scales the contents of the iframe rather than the iframe itself, which harms the impression of this being a zoomed out vie

Yes, that's how it works but it shouldn't work differently as otherwise we'll have scrollbars inside the iframe (so not by the edge). That's particularly important for zoom-out mode scaling and I don't feel like we should be implementing two different ways of scaling the canvas.

**Reply by jameskoster** [reply]
Template parts are looking better in browse view, thanks! There is an issue with the dimensions of the wrapper when you enter edit view though:


https://user-images.githubusercontent.com/846565/200288067-ded41b52-0802-4da7-b015-8d07cc42485d.mp4

> That's particularly important for zoom-out mode scaling and I don't feel like we should be implementing two different ways of scaling the canvas.

I understand, but the problem is that the browser metaphor is broken. For sites that make use of lots of full-width elements the experience will be quite strange. If there's really no way around it on the engineering side then perhaps we should remove the styling on the frame altogether and increase the scale in browse view a bit. This is the result:


https://user-images.githubusercontent.com/846565/200291344-86d1df04-8395-4d33-81e3-37d872790a6f.mp4

The scrollbar is positioned a bit awkwardly, but the document appears the same in both views which seems very important to me.

**Reply by youknowriad** [reply]
> Template parts are looking better in browse view, thanks! There is an issue with the dimensions of the wrapper when you enter edit view though:

This should be fixed.

**Reply by youknowriad** [reply]
> I understand, but the problem is that the browser metaphor is broken. For sites that make use of lots of full-width elements the experience will be quite strange.

The other solution there is to actually show a different background around the scaled area (like we do in zoom-out mode) which means we'll have the "frame" and a visible background/padding within the frame.

**Reply by jameskoster** [reply]
Here's how that would look:


https://user-images.githubusercontent.com/846565/200298815-f71a1d14-9031-4b8c-8152-6ccb90581df2.mp4



I think the metaphor is still broken. The frame boundary is supposed to be synonymous with an actual browser window. If it's not a 1:1 representation then I'd argue there's no point "painting" the frame at all, and no background would be a better approach.

**Reply by jasmussen** [reply]
Would it be possible to ship the branch as an experiment to be activated, or does it touch too many pieces in order for that to be feasible?

**Comment by jasmussen** [comment]
A couple of us from this thread huddled in slack and here are the notes. The main questions were around reaching a baseline for the PR in question that was better than the site editor as it exists today, so that we are able to merge and iterate. 

<details><summary>Notes</summary>

"Browse Mode"

* “Navigation” as an item in the dark sidebar also navigates the content of the frame, as would a “Content” section.

Is it possible to put this effort behind a Gutenberg experiment?

* Possible, could take a day. Probably don't need an experiment for the whole frame, but maybe for specific things like the edit button, or the navigation structure if we want to keep the flow with the W for a bit. 

“Edit” button.

* Since “Edit” is contextual to what you see inside the frame, perhaps it can sit anywhere. 
* We can test some things. It could be anchored with “Design”, it exists at that level. If you’re at a different level, like “Navigation”, maybe there’s a plus there to add menu items. It could also be a higher level. But it’s important to consider the edit button as a semantic element that might exist contextually with different behaviors across other sections.

“The frame”

* Most important piece to get right, we can’t get to a good place unless we adress this.
* It’s content only, no header, no sidebar, no inserter, those exist outside the frame.
* The frame is a specific slot where we assign the contents of the skeleton.
* Technically, once we click edit, the canvas expands, and the toolbar and sidebar appear from outside the canvas. The controls are part of the “Design” interface, and when you’re in fullscreen mode, there’s a toolbar that drops from above.
* A content only view might also help with getting the animation right.
* The editor is an experience, so once we’re in a “zoom out frame mode”, the editor doesn’t have any tools: you’re browsing.
* The header with the W at the top doesn’t have any tools. Once you zoom in, the W/Header area shows contextual tools. It’s just that the frame is now occupying the full page. The tools relate to the state that the full page is in, which is fullscreen mode. 
* The structure is important, that’s why we put so much emphasis on the animation of the canvas, and the UI items. Where does the UI live, conceptually and in the structure of the thing? If the header is inside the frame, it’s not going to look right.
* A header dropping from the top helps convey this.
* The whole page is an editor, and is aware of it, and interacts with its content. You can edit when zoomed out (such as move the navigation around), it’s just that when you zoom in you get more tools available. 

Action items:

* Design a “Templates” subsection of Design. Goal: find out what needs to be “generic” (semantic controls that live in the dark material) and what needs to be local to the view. Does it have a header? Does it show only templates? Does it have a list view that gets loaded when you click “Manage all templates”? 
* Questions to answer: what does the frame contain if we drill in? What are elements of the current view that don’t belong inside the frame? Should the sidebar move? That might help clarify what should be generic and what shouldn’t.

</details>

**Comment by annezazu** [comment]
Hi folks. In testing this out ahead of an FSE Outreach Program exploration, I was struck by a few things that felt quite jarring. To quickly summarize:
- The edit button is wildly small and hard to find. 
- I'm used to being able to enter the site editor and begin editing directly. There's no warning or clarity why I can't.
- The zooming that happens in browse mode doesn't align with zoomed out mode in a way that feels jarring/strange. I wish there was more consistency. 

Here's a video just showing how... strange the experience feels:

https://user-images.githubusercontent.com/26996883/201487951-cce2dec5-5b6f-4431-b7d8-4b8b0741b7f3.mov

I know this is a work in progress but wanted to share early thoughts for now.

**Reply by annezazu** [reply]
As part of the FSE Outreach Program's Site Editor exploration, the [following was brought up as a point of concern that ties in with some of my notes above around the current zooming experience:
](https://make.wordpress.org/test/2022/11/13/fse-program-exploration-site-editor-sneak-peek/#comment-2763)

> The new motions may be too much for some folks. At the risk of adding complexity, has there been discussion of the accessibility aspect of these transition effects?

I've seen some discussion in the PR itself but wanted to note this here to be safe.

**Reply by jasmussen** [reply]
These animations like all others should be subject to the `reduce-motion` user preference, which should disable them.

**Reply by annezazu** [reply]
More feedback has continued to [come in around the edit button/experience:](https://make.wordpress.org/test/2022/11/13/fse-program-exploration-site-editor-sneak-peek/#comment-2765) 

>The edit button is very small/hard to see – I feel like clicking into the content area should allow me to edit. I kept trying to do this (unintentionally)

>My muscle memory made me expect to be able to click on elements to edit them – which I obviously couldn’t do.


I've personally done this as well! It makes me wonder if we need a "double click to edit" UX here too.

**Reply by annezazu** [reply]
>What about being able to double click the canvas window to go into edit mode of the “default” template? As if I had selected a template and clicked the Edit button.

This feedback just came in for the [FSE Program Testing Call #19: Running Through Refinements](https://make.wordpress.org/test/2022/12/14/fse-program-testing-call-19-running-through-refinements/). @jameskoster & @jasmussen curious what you think! The biggest hold up I have mentally is that I know long term browse mode should let you click and see other parts of your site (truly interact) with the edit button being the distinguishing moment when you can edit. I can understand why we might want to preserve true editing until someone selects "edit" if we do allow for a more "live preview" style moving around of the site in time.

**Reply by jasmussen** [reply]
I think that could be a nice power-user feature to potentially explore, but it would likely need to be an affordance built upon an explicit action found in menus or otherwise. Also to be mindful of, a double-tap on tablets means zoom in some cases.

**Comment by annezazu** [comment]
Happy to open this as a separate issue but wanted to note three folks saying the same thing as part of the FSE Outreach Program's exploration around this new feature: there needs to be a more visible way to get back to the dashboard.

From @paaljoachim [here](https://github.com/WordPress/gutenberg/pull/44770#issuecomment-1325729115):

>The < Dashboard link should still be present in the new browse mode, as it gives a clear signal that clicking it will return to the Dashboard. Or Atleast have a Hover over the W (site identity) showing the tooltip that says "Return to Dashboard"

And from two folks in the outreach program:

>I still think that getting back out to the Dashboard could be more obvious. I’ve been working in it for a while now so I know that clicking the W gets you there, but if I was just encountering this for the first time, I would be looking for a “Dashboard” button or link."

& once more:

>Reflecting the sentiment of the previous commenter, it isn’t immediately obvious (to a user new to WordPress, specifically) that the “W” logo is the way to exit the editor."

**Comment by annezazu** [comment]
Two different folks in the FSE Outreach Program's exploration around this feature noted a desire for breadcrumbs of sorts when you're in this new experience. You can hear it discussed in @paaljoachim [video starting here for about 1 minute](https://youtu.be/-UVP8kS_YYQ?t=445) and otherwise here:

>it would be nice to have breadcrumbs indicating what exactly we are editing, as annotated in this screenshot: https://cln.sh/vut5dN

What's the current thinking here? I do think there might be value in emphasizing the breadcrumbs perhaps more. I know we have breadcrumbs at the bottom of the editor but, either way, wanted to note these initial reactions of folks!

Let me know if these need to be issues, @jameskoster! Trying to follow the design discussion since this is still very much a work in progress but happy to open any and all issues.

**Reply by jameskoster** [reply]
Since we aim to add full browse-ability to the zoomed out frame, I think it would be beneficial to display both the URL and the template name in an address-bar style UI:

<img width="1260" alt="Screenshot 2022-11-29 at 11 10 07" src="https://user-images.githubusercontent.com/846565/204514327-d8258710-8f47-4a17-91c2-c50dc9ab58c9.png">

This helps user orientation, and to explain the relationship between content and the various templates. Down the road it can also act as a global search allowing super fast navigation to specific items.

**Reply by jasmussen** [reply]
I'd be happy to try this as a followup to showing the Edit button right aligned in the "Design" slot.

**Reply by jameskoster** [reply]
Yes I don't think it needs to block the PR.

**Reply by annezazu** [reply]
Agreed on all fronts -- this looks like a great iteration and let's get the PR shipped for now :)
